src/reporters/csv.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2017 aixigo AG
+ * Released under the MIT license.
+ * https://github.com/aixigo/arestocats/blob/master/LICENSE
+ */
+
+module.exports = report;
+
+const fs = require( 'fs' );
+const path = require( 'path' );
+const { outcomes } = require( '../outcomes.js' );
+
+const reportsPath = 'reports';
+const resultsPath = path.join( reportsPath, 'results' );
+const csvDir = path.join( resultsPath, 'csv' );
+
+function report( allResults ) {
+   if( !fs.existsSync( reportsPath ) ) {
+      fs.mkdirSync( reportsPath );
+   }
+   if( !fs.existsSync( csvDir ) ) {
+      fs.mkdirSync( csvDir );
+   }
+   allResults.forEach( result => {
+      const testsuites = buildReport( result );
+      testsuites.forEach( summarizeSuite );
+      fs.writeFileSync( path.join( csvDir, `${result.subject.name}.csv` ), asCsv( testsuites ) );
+   } );
+   summarizeCsv( csvDir );
```

Comment:
I'd like to avoid re-reading and -parsing results from disk that we still have in memory at this point.
Summarizing the sum-of-testSuites should be simpler and less bug-prone than summarizing a set of CSV files.

So, the `summarizeCsv` function should be removed and there should be an overall result e.g. computed by running `summarizeSuite` on a synthesized overall suite containing all `testSuites`. This can then be written using `asCsv`.